File: 650-2-keys-keyboard/2-keys-keyboard.py
class Solution:
    def minSteps(self, n: int) -> int:
        if n==1:
            return 0
        else:
            for i in range(n-1,1,-1):
                if n % i == 0:
                    return round(Solution().minSteps(i) + n/i)
        return n

File: 650-2-keys-keyboard/2-keys-keyboard.cpp
class Solution {
public:
    int minSteps(int n) {
        if(n==1){
            return 0;
        }else{
            for(int i=n-1;i>=2;i--){
                if(n%i==0){
                    return minSteps(i)+n/i;
                }
            }
        }
        return n;
    }
};